apis/actions.summerwind.net/v1alpha1/horizontalrunnerautoscaler_types.go

```diff
@@ -114,11 +114,9 @@ type PushSpec struct {
 // to the scale target until ExpirationTime.
 type CapacityReservation struct {
 	Name           string      `json:"name,omitempty"`
+	EffectiveTime  metav1.Time `json:"effectiveTime,omitempty"`
 	ExpirationTime metav1.Time `json:"expirationTime,omitempty"`
 	Replicas       int         `json:"replicas,omitempty"`
-
-	// +optional
-	EffectiveTime metav1.Time `json:"effectiveTime,omitempty"`
```

Comment:
@Nuru May I ask why we changed this in this pull request? What exact issue is this change supposed to solve?

Reply:
Removing "optional" to ensure it set, so that this code works: https://github.com/actions/actions-runner-controller/blob/0b5f9d9b930454e64610374143998c207fcfd977/controllers/actions.summerwind.net/horizontal_runner_autoscaler_batch_scale.go#L181-L182

Remove it if you want to.

Reply:
Thanks for your confirmation. Let me revert this part then!

Whenever you upgrade CRDs in a backward-incompatible way, you should upgrade the controller code first so that it never omits EffectiveTime, so that the controller is forward-compatible!

That's because our current recommended upgrade process is to upgrade CRDs first, and followed by a controller upgrade. Without an extensive explanation of why this is safe, I'd consider any single pull request that modifies both controller and CRDs in a backward-incompatible way lacks consideration into a safe upgrade.

Reply:
And it turned out this `// +optional` never produced an entry under the `required` field of the CRD. Not sure how this has ever worked!
The existence of `// +optional` did not make differences across `make manifests` runs. Odd.